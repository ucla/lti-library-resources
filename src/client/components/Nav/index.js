import React, {useState} from 'react';
import * as constants from '../../constants';

import { AppNav } from '@instructure/ui-navigation';

function Nav({ subjectArea }) {
  const [selectedIndex, setSelectedIndex] = useState(constants.TABS.RESEARCH_GUIDE);

  return (
    <AppNav
      screenReaderLabel="App navigation"
      visibleItemsCount={Object.keys(constants.TABS).length}
    >
      <AppNav.Item
        isSelected={selectedIndex === constants.TABS.RESEARCH_GUIDE}
        renderLabel="Research Guide"
        onClick={() => setSelectedIndex(constants.TABS.RESEARCH_GUIDE)}
      />
      <AppNav.Item
        isSelected={selectedIndex === constants.TABS.COURSE_RESERVES}
        renderLabel="Course Reserves"
        onClick={() => setSelectedIndex(constants.TABS.COURSE_RESERVES)}
      />
      {subjectArea === "CLUSTER" && <AppNav.Item
        isSelected={selectedIndex === constants.TABS.LIBRARY_TOUR}
        renderLabel="Library Tour"
        href="https://spark.adobe.com/page/uJnBMHgK6VMHA/"
        target="_blank"
      />}
      <AppNav.Item
        isSelected={selectedIndex === constants.TABS.RESEARCH_TUTORIALS}
        renderLabel="Research Tutorials"
        href="https://uclalibrary.github.io/research-tips/workshops/"
        target="_blank"
      />
    </AppNav>
  );
}

export default Nav;