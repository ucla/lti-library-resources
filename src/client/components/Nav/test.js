import React from 'react';
import Nav from '.';
import { shallow } from 'enzyme';

it('Tabs render correctly for clusters', () => {
  const wrapper = shallow(
    <Nav subjectArea={"CLUSTER"} />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('Tabs render correctly for non-clusters', () => {
  const wrapper = shallow(
    <Nav subjectArea={"NOT CLUSTER"} />,
  );
  expect(wrapper).toMatchSnapshot();
});