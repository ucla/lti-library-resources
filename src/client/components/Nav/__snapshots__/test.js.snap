// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs render correctly for clusters 1`] = `
<AppNav
  debounce={300}
  elementRef={[Function]}
  margin="0"
  onUpdate={[Function]}
  renderTruncateLabel={[Function]}
  screenReaderLabel="App navigation"
  visibleItemsCount={4}
>
  <Item
    cursor="pointer"
    isDisabled={false}
    isSelected={true}
    onClick={[Function]}
    renderLabel="Research Guide"
  />
  <Item
    cursor="pointer"
    isDisabled={false}
    isSelected={false}
    onClick={[Function]}
    renderLabel="Course Reserves"
  />
  <Item
    cursor="pointer"
    href="https://spark.adobe.com/page/uJnBMHgK6VMHA/"
    isDisabled={false}
    isSelected={false}
    onClick={[Function]}
    renderLabel="Library Tour"
    target="_blank"
  />
  <Item
    cursor="pointer"
    href="https://uclalibrary.github.io/research-tips/workshops/"
    isDisabled={false}
    isSelected={false}
    onClick={[Function]}
    renderLabel="Research Tutorials"
    target="_blank"
  />
</AppNav>
`;

exports[`Tabs render correctly for non-clusters 1`] = `
<AppNav
  debounce={300}
  elementRef={[Function]}
  margin="0"
  onUpdate={[Function]}
  renderTruncateLabel={[Function]}
  screenReaderLabel="App navigation"
  visibleItemsCount={4}
>
  <Item
    cursor="pointer"
    isDisabled={false}
    isSelected={true}
    onClick={[Function]}
    renderLabel="Research Guide"
  />
  <Item
    cursor="pointer"
    isDisabled={false}
    isSelected={false}
    onClick={[Function]}
    renderLabel="Course Reserves"
  />
  <Item
    cursor="pointer"
    href="https://uclalibrary.github.io/research-tips/workshops/"
    isDisabled={false}
    isSelected={false}
    onClick={[Function]}
    renderLabel="Research Tutorials"
    target="_blank"
  />
</AppNav>
`;
